src/core.rs

```diff
@@ -393,6 +393,14 @@ pub fn run_core() -> Subscription<Message> {
                         let db_path = path.join("harbor.sqlite");
 
                         let db_path = db_path.to_str().unwrap().to_string();
+
+                        // if the db file doesn't exist, call setup_db
+                        if !std::path::Path::new(&db_path).exists() {
+                            if let Err(e) = setup_db(&db_path, password.clone()) {
```

Comment:
this makes it so we call setup_db twice. should instead just not call `check_password` if the database file does not exist

Reply:
I searched the whole code base for setup_db, and it's not being called anywhere other than in the setup_test_db; maybe I am missing something

Reply:
being called on line 419 here

Reply:
It is done, tested, and working as expected.